src/main/java/pl/mmieczak/springbootexercise_22_1/UserRepository.java

```diff
@@ -22,4 +24,40 @@ public UserRepository() {
     public void add(String name, String surname, int age) {
         userCollection.add(new User(name, surname, age));
     }
+
+    public void removeAll(Stream<User> streamToRemove) {
+        userCollection.removeAll(streamToRemove.collect(Collectors.toList()));
```

Comment:
Tutaj widziałbym metodę w której podajemy 3 parametry (imie, nazwisko i wiek) a ta metoda decyduje kogo usunąć, a nie że musimy jej podać kogo ma usunąć.

Reply:
stwierdziłem, że skoro mam metodę która mi już odfiltrowała dane i mam je na widelcu to lepiej puścić execute, cała logika jest w funkcji filterDataUsingFormValues(name, surname, age), ale samo filtrowanie mogłem uruchomić w kolejnej metodzie do usuwania, to by było czytelniejsze?

Reply:
Tak, tylko że lepiej jest np zrobić w takim razie metodę `delete` z tymi trzema parametrami która pod spodem użyje sobie motody `filterDataUsing...` a potem wykona `removeAll` :) Chodzi o to żeby ukrywać pod jednoznacznie brzmiącymi metodami bardziej skomplikowaną logikę, żeby nikt nie musiał pamiętać żeby wywoływać metodę `filterDataUsing...` a potem `removeAll`